src/main/java/server/Server.java

```diff
@@ -0,0 +1,157 @@
+package server;
+
+import commons.requests.DeleteBookingRequest;
+import commons.requests.QueryAvailabilityRequest;
+import commons.requests.Request;
+
+import commons.requests.*;
+import commons.responses.*;
+import commons.responses.Response;
+import commons.responses.TestResponse;
+import commons.rpc.ServerCommunicator;
+import commons.utils.ClientRequest;
+import javafx.util.Pair;
+import server.entities.*;
+import server.managers.*;
+import server.translator.Translator;
+
+
+import java.net.InetAddress;
+import java.util.Hashtable;
+import java.util.Optional;
+
+//TODO implement bookId count
+//TODO remove client id from bookings(?)
+
+public class Server {
+    BookingManager bookingManager;
+    ObserverManager observerManager;
+    Hashtable<String, Facility> facilTable;
+    ServerCommunicator serverCommunicator;
+    Translator translator;
+
+    int testCounter = 0;
+    int bookingIdCounter;
+
+
+    public Server(int serverPort, boolean atMostOnce) {
+        this.bookingManager = new BookingManager();
+        this.observerManager = new ObserverManager();
+        this.facilTable = new Hashtable<>();
+        this.serverCommunicator = new ServerCommunicator(serverPort, atMostOnce);
+        this.translator = new Translator(serverCommunicator);
+
+        bookingIdCounter =0;
+    }
+
+    public static void main(String[] args) {
+        int serverPort = Integer.parseInt(args[0]);
+        boolean atMostOnce = Boolean.parseBoolean(args[1]);
+        Server server = new Server(serverPort, atMostOnce);
+        while (true) {
+            server.run(0);
+        }
+    }
+
+    void run(int timeout) {
+        Optional<ClientRequest> optionalClientRequest;
+        if (timeout == 0){
+            optionalClientRequest = serverCommunicator.receive();
+        }
+        else{
+            optionalClientRequest = serverCommunicator.receive(timeout);
+        }
+
+        if (optionalClientRequest.isPresent()) {
+            ClientRequest clientRequest = optionalClientRequest.get();
+            Request request = clientRequest.request;
+            Response response = null;
+            String facilName = null;
+            Exception e = null;
+
+            //TODO
+            // Run tests to see if type castings (eg. BookingFacilityResponse to Response) is valid
```

Comment:
Don't think theres a need to cast since it extends Response

Reply:
i see, yea my IDE keep showing me redlines so i thought i needed to cast